proxy_worker/dispatcher.py

```diff
@@ -0,0 +1,528 @@
+            category=record.name,
+            log_category=log_category
+        )
+
+        invocation_id = get_current_invocation_id()
+        if invocation_id is not None:
+            log['invocation_id'] = invocation_id
+
+        self._grpc_resp_queue.put_nowait(
+            protos.StreamingMessage(
+                request_id=self.request_id,
+                rpc_log=protos.RpcLog(**log)))
+
+    @property
+    def request_id(self) -> str:
+        return self._request_id
+
+    @property
+    def worker_id(self) -> str:
+        return self._worker_id
+
+    @classmethod
+    async def connect(cls, host: str, port: int, worker_id: str,
+                      request_id: str, connect_timeout: float):
+        loop = asyncio.events.get_event_loop()
+        disp = cls(loop, host, port, worker_id, request_id, connect_timeout)
+        # Safety check for mypy
+        if disp._grpc_thread is not None:
+            disp._grpc_thread.start()
+        await disp._grpc_connected_fut
+        logger.info('Successfully opened gRPC channel to %s:%s ', host, port)
+        return disp
+
+    def __poll_grpc(self):
+        options = []
+        if self._grpc_max_msg_len:
+            options.append(('grpc_local.max_receive_message_length',
+                            self._grpc_max_msg_len))
+            options.append(('grpc_local.max_send_message_length',
+                            self._grpc_max_msg_len))
+
+        channel = grpc.insecure_channel(
+            f'{self._host}:{self._port}', options)
+
+        try:
+            grpc.channel_ready_future(channel).result(
+                timeout=self._grpc_connect_timeout)
+        except Exception as ex:
+            self._loop.call_soon_threadsafe(
+                self._grpc_connected_fut.set_exception, ex)
+            return
+        else:
+            self._loop.call_soon_threadsafe(
+                self._grpc_connected_fut.set_result, True)
+
+        stub = protos.FunctionRpcStub(channel)
+
+        def gen(resp_queue):
+            while True:
+                msg = resp_queue.get()
+                if msg is self._GRPC_STOP_RESPONSE:
+                    grpc_req_stream.cancel()
+                    return
+                yield msg
+
+        grpc_req_stream = stub.EventStream(gen(self._grpc_resp_queue))
+        try:
+            for req in grpc_req_stream:
+                self._loop.call_soon_threadsafe(
+                    self._loop.create_task, self._dispatch_grpc_request(req))
+        except Exception as ex:
+            if ex is grpc_req_stream:
+                # Yes, this is how grpc_req_stream iterator exits.
+                return
+            error_logger.exception(
+                'unhandled error in gRPC thread. Exception: {0}'.format(
+                    ''.join(traceback.format_exception(ex))))
+            raise
+
+    async def _dispatch_grpc_request(self, request):
+        content_type = request.WhichOneof("content")
+
+        match content_type:
```

Comment:
Instead of if-else, use a dictionary

Reply:
So something like

```python
handlers = handlers["content_type"]
```

And handlers is a dict [<content_type:str>, <func_point:addr>]